src/clj/rems/db/attachments.clj

```diff
@@ -25,6 +29,42 @@
      :attachment/data data
      :attachment/type type}))
 
+(defn scan-for-malware
+  "Feeds byte-array to STDIN and runs executable at malware-scanner-path returns true if malware executable returns a non-zero status-code, false otherwise, logs STERR of executable"
+  [malware-scanner-path byte-array]
+  (let [scan-output (sh "sh" "-c" malware-scanner-path :in byte-array)]
+    {:detected (not= (:exit scan-output) 0)
+     :log (:err scan-output)}))
```

Comment:
Might some malware scanner output to stdout i.e. `:out`? Should we `:log` both of them?

Reply:
I've put this note in the documentation:

>Output written to `STDOUT` is discarded for now, a future implementation might use this data stream to communicate scan progress to the user.

Leaving this channel open for the above purpose, will ensure that REMS can support that feature in the future while ensuring backwards compatibility and agnosticism to malware scanner implementation. That was at least my reasoning.

Reply:
Also, I believe that it is sound to expect: If a program exits with non-zero status code, what is printed to `STDERR` relates to the reasoning for the non-zero status code.

Reply:
@Macroz let me know if you agree with the above, otherwise I'll implement the suggestion so we can resolve this last conversation and get this merged.

Reply:
I don't think it is good to leave STDOUT out from the logs. It will be quite impossible to know from production what happened when something goes wrong. I also find it likely that different scanners may or may not print to STDERR, or respect the status codes. So best would be to log both.

Have you tested this with some real malware scanner already?

Reply:
> I don't think it is good to leave STDOUT out from the logs. It will be quite impossible to know from production what happened when something goes wrong. I also find it likely that different scanners may or may not print to STDERR, or respect the status codes. So best would be to log both.

Makes sense, I guess you should still be able to communicate both progress and result over STDOUT in the future. 

> Have you tested this with some real malware scanner already?

I have not. I will once this issue is resolved.

Reply:
Tested with [clamscan](https://docs.clamav.net/manual/Usage/Scanning.html#clamscan). 

Pointed `scanner-path` to a shellscript with the following content:

```
#!/bin/sh

clamscan -

```

Given a known safe file, the file uploaded correctly without issues and logged the following in the application log:


```
2023-11-07 13:55:34,415 [jetty-61 rqc:1 alice sKpEy67R] INFO  rems.middleware - req > :post /api/applications/add-attachment?application-id=15
2023-11-07 13:55:34,429 [jetty-61 rqc:1 alice sKpEy67R] INFO  rems.middleware - > :post /api/applications/add-attachment?application-id=15 lang: :en user: {:userid alice, :name Alice Applicant, :email alice@example.com, :nickname In Wonderland, :organizations [{:organization/id default}], :groups [VO1 B 1 role=R], :researcher-status-by so}  roles: #{:applicant :logged-in}
2023-11-07 13:55:34,436 [jetty-61 rqc:1 alice sKpEy67R] INFO  rems.api.applications - > params: {:application-id 15, :file {:filename notavirus.txt, :content-type text/plain, :tempfile ring-multipart-2679007160370982202.tmp, :size 17}}
2023-11-07 13:55:46,897 [jetty-61 rqc:1 alice sKpEy67R] INFO  rems.db.attachments - stdin: OK

----------- SCAN SUMMARY -----------
Known viruses: 8677712
Engine version: 0.103.9
Scanned directories: 0
Scanned files: 1
Infected files: 0
Data scanned: 0.00 MB
Data read: 0.00 MB (ratio 0.00:1)
Time: 12.436 sec (0 m 12 s)
Start Date: 2023:11:07 13:55:34
End Date:   2023:11:07 13:55:46

2023-11-07 13:55:46,911 [jetty-61 rqc:1 alice sKpEy67R] INFO  rems.middleware - req < :post /api/applications/add-attachment?application-id=15 200 12495ms 
```

Used [this test file](https://gist.github.com/mikecastrodemaria/0843f8828fef7c60558a58248fcb724c), which resulted in the upload being rejected with a 'bad request' error message on the frontend, and the following logged:

```
2023-11-07 13:48:42,280 [jetty-54 rqc:1 alice mzrKDr79] INFO  rems.middleware - > :post /api/applications/add-attachment?application-id=15 lang: :en user: {:userid alice, :name Alice Applicant, :email alice@example.com, :nickname In Wonderland, :organizations [{:organization/id default}], :groups [VO1 B 1 role=R], :researcher-status-by so}  roles: #{:applicant :logged-in}
2023-11-07 13:48:42,287 [jetty-54 rqc:1 alice mzrKDr79] INFO  rems.api.applications - > params: {:application-id 15, :file {:filename test-doc-see-as-virus-by-clamav.txt, :content-type text/plain, :tempfile ring-multipart-1256581554190874369.tmp, :size 68}}
2023-11-07 13:48:55,183 [jetty-54 rqc:1 alice mzrKDr79] INFO  rems.db.attachments - stdin: Win.Test.EICAR_HDB-1 FOUND

----------- SCAN SUMMARY -----------
Known viruses: 8677712
Engine version: 0.103.9
Scanned directories: 0
Scanned files: 1
Infected files: 1
Data scanned: 0.00 MB
Data read: 0.00 MB (ratio 0.00:1)
Time: 12.873 sec (0 m 12 s)
Start Date: 2023:11:07 13:48:42
End Date:   2023:11:07 13:48:55

2023-11-07 13:48:55,186 [jetty-54 rqc:1 alice mzrKDr79] INFO  rems.api - bad-request Malware detected
2023-11-07 13:48:55,192 [jetty-54 rqc:1 alice mzrKDr79] INFO  rems.middleware - req < :post /api/applications/add-attachment?application-id=15 400 12924ms 
```

So I reckon it works!